src/Server.kt

```diff
@@ -197,16 +199,39 @@ fun Application.module() {
 
                     if (call.parameters.contains("simulationIndex")) {
                         val simIndex = call.parameters["simulationIndex"]?.toInt() ?: 0
```

Comment:
If the run method is simplified with a single *simulation* parameter then this block can be simplified by getting the simulation at *i*:
```kotlin
val simulation = model.simulations[call.parameters["simulationIndex"]?.toInt() ?: 0]
call.respondText(model.run(simulation))
```

When the passed index is missing it is similar but at index 0.

Reply:
done